sat_vapor_pres/include/sat_vapor_pres_k.inc

```diff
@@ -1128,893 +1009,1202 @@
+ real(kind=FMS_SVP_KIND_), intent(out), dimension(:,:)  :: esat, desat
  integer, intent(out)               :: nbad
 
  real    :: tmp, del
  integer :: ind, i, j
 
+ real(kind=FMS_SVP_KIND_) :: dtres_loc, tepsl_loc, tminl_loc,  dtinvl_loc
+
+   dtres_loc=real(dtres, FMS_SVP_KIND_)
+   tminl_loc=real(tminl, FMS_SVP_KIND_)
+   tepsl_loc=real(tepsl, FMS_SVP_KIND_)
+   dtinvl_loc=real(dtinvl, FMS_SVP_KIND_)
+
+
    nbad = 0
    do j = 1, size(temp,2)
    do i = 1, size(temp,1)
-     tmp = temp(i,j)-tminl
-     ind = int(dtinvl*(tmp+tepsl))
+     tmp = temp(i,j)-tminl_loc
+     ind = int(dtinvl_loc*(tmp+tepsl_loc))
      if (ind < 0 .or. ind >= table_siz)  then
        nbad = nbad+1
      else
-       del = tmp-dtres*real(ind)
-       esat(i,j) = TABLE(ind+1) + &
-                   del*(DTABLE(ind+1) + del*D2TABLE(ind+1))
-       desat(i,j) = DTABLE(ind+1) + 2.*del*D2TABLE(ind+1)
+       del = tmp-dtres_loc*real(ind)
+       esat(i,j) = real(TABLE(ind+1),FMS_SVP_KIND_) &
+                 + del*( real(DTABLE(ind+1),FMS_SVP_KIND_) &
+                               + del*real(D2TABLE(ind+1),FMS_SVP_KIND_) )
+       desat(i,j) = real(DTABLE(ind+1),FMS_SVP_KIND_) &
+                  + real(2.0_r8_kind,FMS_SVP_KIND_)*del*real(D2TABLE(ind+1),FMS_SVP_KIND_)
      endif
    enddo
    enddo
 
- end subroutine lookup_es_des_k_2d
+ end subroutine LOOKUP_ES_DES_K_2D_
 
 !#######################################################################
 
- subroutine lookup_es_des_k_1d (temp, esat, desat, nbad)
- real, intent(in),  dimension(:)  :: temp
- real, intent(out), dimension(:)  :: esat, desat
+ subroutine LOOKUP_ES_DES_K_1D_ (temp, esat, desat, nbad)
+ real(kind=FMS_SVP_KIND_), intent(in),  dimension(:)  :: temp
+ real(kind=FMS_SVP_KIND_), intent(out), dimension(:)  :: esat, desat
  integer, intent(out)             :: nbad
 
- real    :: tmp, del
+ real(kind=FMS_SVP_KIND_) :: tmp, del
  integer :: ind, i
 
+ real(kind=FMS_SVP_KIND_) :: dtres_loc, tepsl_loc, tminl_loc,  dtinvl_loc
+
+   dtres_loc=real(dtres, FMS_SVP_KIND_)
+   tminl_loc=real(tminl, FMS_SVP_KIND_)
+   tepsl_loc=real(tepsl, FMS_SVP_KIND_)
+   dtinvl_loc=real(dtinvl, FMS_SVP_KIND_)
+
+
    nbad = 0
    do i = 1, size(temp,1)
-     tmp = temp(i)-tminl
-     ind = int(dtinvl*(tmp+tepsl))
+     tmp = temp(i)-tminl_loc
+     ind = int(dtinvl*(tmp+tepsl_loc))
      if (ind < 0 .or. ind >= table_siz)  then
        nbad = nbad+1
      else
-       del = tmp-dtres*real(ind)
-       esat(i) = TABLE(ind+1) + &
-                   del*(DTABLE(ind+1) + del*D2TABLE(ind+1))
-       desat(i) = DTABLE(ind+1) + 2.*del*D2TABLE(ind+1)
+       del = tmp-dtres_loc*real(ind,FMS_SVP_KIND_)
+       esat(i) = real(TABLE(ind+1),FMS_SVP_KIND_) &
+               + del*( real(DTABLE(ind+1),FMS_SVP_KIND_) &
+                                + del*real(D2TABLE(ind+1),FMS_SVP_KIND_) )
+       desat(i) = real(DTABLE(ind+1),FMS_SVP_KIND_) &
+               + real(2.0_r8_kind,FMS_SVP_KIND_)*del*real(D2TABLE(ind+1),FMS_SVP_KIND_)
```

Comment:
Same 2.0

Reply:
2.0 could be single precision (if compiled with real-size-32) and converting it to r8_kind could lead to "misrepresentation" of 2.0, such as 2.00000000000001.  Perhaps  it would be best to play it safe and use the double precision number to convert to r4/r8_kind.